examples/coappearances.cpp

```diff
@@ -0,0 +1,327 @@
+    }
+
+    auto major = std::stoul( s.substr( 0, pos ) );
+    if ( ( 1 << co::Chapter::MajorType::bit_width ) <= major )
+    {
+        throw std::runtime_error( "Major chapter overflow" );
+    }
+
+    auto minor = std::stoul( s.substr( pos + 1 ) );
+    if ( ( 1 << co::Chapter::MinorType::bit_width ) <= minor )
+    {
+        throw std::runtime_error( "Major chapter overflow" );
+    }
+
+    return {static_cast< uint8_t >( major ), static_cast< uint8_t >( minor )};
+}
+
+void
+convert_characters( const picojson::object& characters,
+                    const CharactersIndex& characters_index,
+                    co::GraphBuilder builder,
+                    std::string& strings )
+{
+    auto vertices = builder.start_vertices( );
+    auto vertices_data = builder.start_vertices_data( );
+
+    for ( const auto& kv : characters )
+    {
+        const auto& id = kv.first;
+        const auto& data = kv.second.get< picojson::object >( );
+
+        auto character = vertices.grow( );
+        character.name_ref = strings.size( );
+        strings += data.at( "name" ).get< std::string >( ) + '\0';
+
+        if ( data.count( "nickname" ) )
+        {
+            auto nickname = vertices_data.add_to_current_item< co::Nickname >( );
+            nickname.ref = strings.size( );
+            strings += data.at( "nickname" ).get< std::string >( ) + '\0';
+        }
+
+        if ( data.count( "description" ) )
+        {
+            auto description = vertices_data.add_to_current_item< co::Description >( );
+            description.ref = strings.size( );
+            strings += data.at( "description" ).get< std::string >( ) + '\0';
+        }
+
+        if ( data.count( "relation" ) )
+        {
+            const auto& relation = data.at( "relation" ).get< picojson::object >( );
+            if ( relation.at( "to" ).is< std::string >( ) )
+            {
+                auto rel = vertices_data.add_to_current_item< co::UnaryRelation >( );
+                rel.kind_ref = strings.size( );
+                strings += relation.at( "kind" ).get< std::string >( ) + '\0';
+                rel.to_ref = characters_index.at(
+                    convert_id( relation.at( "to" ).get< std::string >( ) ) );
+            }
+            else
+            {
+                auto rel = vertices_data.add_to_current_item< co::BinaryRelation >( );
+                rel.kind_ref = strings.size( );
+                strings += relation.at( "kind" ).get< std::string >( ) + '\0';
+                auto to_refs = relation.at( "to" ).get< picojson::array >( );
+                assert( to_refs.size( ) == 2 );
+                rel.to_a_ref
+                    = characters_index.at( convert_id( to_refs[ 0 ].get< std::string >( ) ) );
+                rel.to_a_ref
+                    = characters_index.at( convert_id( to_refs[ 1 ].get< std::string >( ) ) );
+            }
+        }
+
+        vertices_data.next_item( );
```

Comment:
This hit me several times already. In `vector` I need first to call `grow` and then start adding data. In the multivector it is the other way around: first, add the data (to the bucket), then call `next_item`. What do you think about unifying the behavior? 

We could even go further and unify the interface by replacing `next_item` by `grow`.

Reply:
Solved in #25.